### PR TITLE
Zeppelin Solidity Hot Loader integration for ZepKit

### PR DESCRIPTION
**Pull request by ylv-io**
PR adds support for hot reloading of `.sol` files using webpack loader in order to speed up development and improve developer experience.

Features
* Allows importing `.sol` files directly into source code.
* Converts `.sol` files into `.json` using `compile`->`zos push`->`zos update`.
* Uses only development network, so won't run on any other networks.
* You can set the development network name using loader's config.
* Pulls build directory from truffle config.
* In case zos command not available, uses cached versions of .json files.
* Reports error to console output and dev tools.
* Compilation can be disabled using loader's disabled config.

[Zeppelin Solidity Hot Loader](https://github.com/ylv-io/zeppelin-solidity-hot-loader) is located in the separate GitHub repo, yet it is part of the PR because it is used as a loader in ZepKit.  [NPM package](https://www.npmjs.com/package/zeppelin-solidity-hot-loader) is published to enable `npm install command`.

Changes
* react-app-rewired package added in order to inject loader into CRA. config-overrides.js contains the loader config.
* ModuleScopePlugin disabled in order to allow import `.sol` files outside `src` directory.
* `require('Contract.json')` replaced with `require('Contract.sol')`
* Instructions page supplemented with additional instructions for Hot Reloading.
* truffle package removed due to the fact that ZepKit install instructions tell to run `npm install -g truffle@5.0.2 && npm install -g ganache-cli@6.3.0`

### CODE REVIEW

File: client/src/components/Instructions/index.js [hunk 1]
```diff
@@ -266,6 +266,33 @@ export default class Instructions extends Component {
             Reload
           </Button>
         </div>
+        <h2> Upgrading on Development Network </h2>
```
Comment (rrecuero) [inline client/src/components/Instructions/index.js:276]: I would wrap all this section and only show it if it is local ganache. If it is, hide the button above and show these instructions

File: client/src/components/Instructions/index.js [hunk 2]
```diff
@@ -266,6 +266,33 @@ export default class Instructions extends Component {
...
+          </div>
+          <div className={styles.code}>
+            <code>
+              {`// function decreaseCounter(uint256 amount) public returns (bool) {`}
+            </code>
+          </div>
+        </div>
+        <div className={styles.step}>
+          <div className={styles.instruction}>
+            2. Save the changes and wait for .sol files compilation, ZOS commands automatically executed.
```
Comment (rrecuero) [inline client/src/components/Instructions/index.js]: I'd rephrase this step and remove the step below
"and wait for the .sol files to compile. Upon completion, ZOS will push and update your smart contracts "

File: client/src/components/Instructions/index.js (hunk 1 above)
Comment (rrecuero) [reply client/src/components/Instructions/index.js:276]: otherwise show the normal ones

**Comment by ylv-io** [comment]
Two more features added Solidity Hot Loader:
* Allows to specify contract name for .sol files, like require("Contract.sol?contract=Counter").
* Tracks dependencies for contracts, so children are updated when parents are modified.

Instructions on upgrading smart contract are now dynamic based on network type and loader state (disabled/enabled).
